### PR TITLE
= - Minor cleanups, will get an appropriate commit message soon

### PR DESCRIPTION
**Pull request by viktorklang**
Do not merge

w.r.t. Module.toString, here's a comparison:

before:

``` scala
scala> Flow[Int].map(_ * 2).filter(_ != 2)
res0: akka.stream.scaladsl.Flow[Int,Int,akka.NotUsed] =
Flow(
 Module: unnamed
 Modules:
   map
   akka.stream.impl.StreamLayout$CopiedModule
 Downstreams:
   map.out -> filter.in
 Upstreams:
   filter.in -> map.out
)
```

after:

``` scala
scala> Flow[Int].map(_ * 2).filter(_ != 2)
res0: akka.stream.scaladsl.Flow[Int,Int,akka.NotUsed] =
Flow(
  Name: unnamed
  Modules:
    map
    filter (copy)
  Downstreams:
    map.out -> filter.in
  Upstreams:
    filter.in -> map.out
)
```

### CODE REVIEW

File: akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteRestartedQuarantinedSpec.scala [hunk 1]
```diff
@@ -46,7 +46,7 @@ object RemoteRestartedQuarantinedSpec extends MultiNodeConfig {
   class Subject extends Actor {
     def receive = {
       case "shutdown" ⇒ context.system.terminate()
-      case "identify" ⇒ sender() ! (AddressUidExtension(context.system).addressUid, self)
+      case "identify" ⇒ sender() ! (AddressUidExtension(context.system).addressUid -> self)
```
Comment (drewhk) [inline akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteRestartedQuarantinedSpec.scala]: thanks

File: akka-stream/src/main/scala/akka/stream/impl/Stages.scala [hunk 2]
```diff
@@ -98,8 +99,8 @@ private[stream] object Stages {
 
     val subscriberSink = name("subscriberSink")
     val cancelledSink = name("cancelledSink")
-    val headSink = name("headSink") and inputBuffer(initial = 1, max = 1)
-    val headOptionSink = name("headOptionSink") and inputBuffer(initial = 1, max = 1)
+    val headSink = name("headSink") and inputBufferOne
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala]: nice

**Comment by drewhk** [comment]
can you add the proper toString to CopiedModule in the meantime?

**Comment by drewhk** [comment]
LGTM so far

File: akka-stream/src/main/scala/akka/stream/impl/Stages.scala (hunk 2 above)
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala]: :)

**Comment by viktorklang** [comment]
@drewhk Sure, I can do something to CopiedModule, do we just want to delegate to the copy or should we decorate it?

**Comment by drewhk** [comment]
It depends whom we target with toString. Since it is the default "print this to me" method on the JVM I would say that people should not learn about the existence of copied modules since that is not their concern. We can add a "toDebugString" method for our convenience very easily if we need it later.

**Comment by viktorklang** [comment]
@drewhk

Problem is this in CompositeModule: `Modules: ${subModules.iterator.map(m ⇒ "\n   " + m.attributes.nameOrDefault(m.getClass.getName)).mkString("")}`

~~~scala
    override def toString =
      s"""
        | Module: ${this.attributes.nameOrDefault("unnamed")}
        | Modules: ${subModules.iterator.map(m ⇒ "\n   " + m.attributes.nameOrDefault(m.getClass.getName)).mkString("")}
        | Downstreams: ${downstreams.iterator.map { case (in, out) ⇒ s"\n   $in -> $out" }.mkString("")}
        | Upstreams: ${upstreams.iterator.map { case (out, in) ⇒ s"\n   $out -> $in" }.mkString("")}
        |""".stripMargin
~~~

Not sure What the proper fix is. do m.toString iso m.getClass.getName?

File: akka-stream/src/main/scala/akka/stream/impl/Stages.scala [hunk 3]
```diff
@@ -23,6 +23,7 @@ private[stream] object Stages {
 
   object DefaultAttributes {
     val IODispatcher = ActorAttributes.Dispatcher("akka.stream.default-blocking-io-dispatcher")
```
Comment (vasseurjb) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala]: Codacy Issue found: [Field name 'IODispatcher' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/vasseurjb/akka/file/2046622393/issues/source?bid=3132306&fileBranchId=3132369#l25)

File: akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala [hunk 4]
```diff
@@ -380,11 +382,12 @@ object StreamLayout {
...
     override def toString =
       s"""
-        | Module: ${this.attributes.nameOrDefault("unnamed")}
-        | Modules: ${subModules.iterator.map(m ⇒ "\n   " + m.attributes.nameOrDefault(m.getClass.getName)).mkString("")}
-        | Downstreams: ${downstreams.iterator.map { case (in, out) ⇒ s"\n   $in -> $out" }.mkString("")}
-        | Upstreams: ${upstreams.iterator.map { case (out, in) ⇒ s"\n   $out -> $in" }.mkString("")}
-        |""".stripMargin
+         |  Module: ${this.attributes.nameOrDefault("unnamed")}
+         |  Modules:
+         |    ${subModules.iterator.map(m ⇒ m.attributes.nameLifted.getOrElse(m.toString.replaceAll("\n", "\n    "))).mkString("\n    ")}
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: The idea here is to defer the toString to only occur if there is no `name`, this code is identical to FusedModule below.

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: Also note that it doesn't fall back to the modules' class' name, but to its String representation.

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 5]
```diff
@@ -25,13 +25,17 @@ import scala.util.Try
...
-  def carbonCopy: Module = CopiedModule(shape.deepCopy(), Attributes.none, this)
+  override def carbonCopy: Module = CopiedModule(shape.deepCopy(), Attributes.none, this)
 
-  def replaceShape(s: Shape): Module = CompositeModule(this, s)
+  override def replaceShape(s: Shape): Module =
+    if (s != shape) CompositeModule(this, s)
+    else this
 
-  def subModules: Set[Module] = Set.empty
+  override def subModules: Set[Module] = Set.empty
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: `override` all the things!

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 6]
```diff
@@ -25,13 +25,17 @@ import scala.util.Try
...
 
-  def withAttributes(attributes: Attributes): Module = new GraphStageModule(shape, attributes, stage)
+  override def withAttributes(attributes: Attributes): Module = new GraphStageModule(shape, attributes, stage)
+
+  override def toString: String = stage.toString
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: One can argue about the correctness of this, but at the very least, it is better than the default for readability.

File: akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala [hunk 7]
```diff
@@ -345,34 +343,38 @@ object StreamLayout {
...
+        shape.requireSamePortsAs(s)
+        CompositeModule(this, s)
+      } else this
 
     override val materializedValueComputation: MaterializedValueNode = Atomic(copyOf)
 
     override def isCopied: Boolean = true
 
-    override def toString: String = "copy of " + copyOf.toString
+    override def toString: String = s"$copyOf (copy)"
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala]: This was a bit of stroke of simplicity

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 8]
```diff
@@ -25,13 +25,19 @@ import scala.util.Try
...
 
-  def subModules: Set[Module] = Set.empty
+  override def subModules: Set[Module] = Set.empty
 
-  def withAttributes(attributes: Attributes): Module = new GraphStageModule(shape, attributes, stage)
+  override def withAttributes(attributes: Attributes): Module =
+    if (attributes ne this.attributes) new GraphStageModule(shape, attributes, stage)
+    else this
+
+  override def toString: String = stage.toString
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Perhaps simplistic, but I found it useful.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: yeah, they usually have a proper toString defined.

**Comment by rkuhn** [comment]
Thanks @viktorklang !

**Pull request merged**